server/src/main/java/org/elasticsearch/action/admin/cluster/stats/CCSTelemetrySnapshot.java

```diff
@@ -0,0 +1,376 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.cluster.stats;
+
+import org.elasticsearch.action.admin.cluster.stats.LongMetric.LongMetricValue;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Holds a snapshot of the CCS telemetry statistics from {@link CCSUsageTelemetry}.
+ * Used to hold the stats for a single node that's part of a {@link ClusterStatsNodeResponse}, as well as to
+ * accumulate stats for the entire cluster and return them as part of the {@link ClusterStatsResponse}.
+ */
+public final class CCSTelemetrySnapshot implements Writeable, ToXContentFragment {
+    private long totalCount;
+    private long successCount;
+    private final Map<String, Long> failureReasons;
+
+    private final LongMetricValue took;
+    private final LongMetricValue tookMrtTrue;
+    private final LongMetricValue tookMrtFalse;
+    private long remotesPerSearchMax;
+    private double remotesPerSearchAvg;
+    private long skippedRemotes;
+
+    private final Map<String, Long> featureCounts;
+
+    private final Map<String, Long> clientCounts;
+    private final Map<String, PerClusterCCSTelemetry> byRemoteCluster;
+
+    public long getTotalCount() {
+        return totalCount;
+    }
+
+    public long getSuccessCount() {
+        return successCount;
+    }
+
+    public Map<String, Long> getFailureReasons() {
+        return Collections.unmodifiableMap(failureReasons);
+    }
+
+    public LongMetricValue getTook() {
+        return took;
+    }
+
+    public LongMetricValue getTookMrtTrue() {
```

Comment:
Somewhere in this class can you put a code comment as to what "mrt" means. People not familiar with CCS will not know what that refers to.